crate_universe/tests/cargo_integration_test.rs

```diff
@@ -295,3 +295,67 @@ fn feature_generator_crate_combined_features() {
         json!(["default", "derive", "serde_derive", "std"])
     );
 }
+
+#[test]
+fn feature_generator_crate_optional_deps_disabled() {
```

Comment:
Is this something that can be unit tested instead? I don't know that this feature *needs* to be integration tested since all the info exists in metadata?